### ISSUE TITLE
Review `check_quarter_format` function

### ISSUE THREAD

**Issue by Jennit07** [Public-Health-Scotland/source-linkage-files#886]
This caused us issues when trying to run the individual file in the december update. Making a note to go back and review this - i have commented out the function for running purposes. See the script `get_fy_quarter_dates.R`

https://github.com/Public-Health-Scotland/source-linkage-files/blob/dec-update-23/R/get_fy_quarter_dates.R

**Comment by Jennit07** [comment]
requires investigation - decide if we actually need this as in our update we just excluded this for running

**Comment by Jennit07** [comment]
The function `check_quarter_format` is not needed anymore as we have `period_start_date` and `period_end_date` which is being read in from the SC platform. These can be used in place of calculations for the start and end of the quarter. 

I have found that most quarter calculation functions are not needed. I have removed this and used `period_start_date` or `period_end_date` in place for the CH processing script which looked like it was the only script using these. I have made a few changes but also need to look more into this. The function `calculate_stay` is using some quarter dates for calculation in care home data but this could be amended to use `period_end_date`. I need to look at the impact of removing the quarter functions all together as it is a little bit confusing with social care code in the mix and how we deal with missing end dates. 

There is one line being used in `create_individual_file` which is impacted by this but when i was looking at the code, at this stage in processing `period_start_date` or `period_end_date` are not available as they are dropped in the process. We could use `sc_latest_submission` but that would be `period`, e.g. the format 2018Q2 which is not always reliable as we know some areas only submit annual data. Needs more investigation, especially to get the logic consistent with social care code.
